pandas/_libs/tslibs/meson.build

```diff
@@ -0,0 +1,75 @@
+# TODO: can this be removed, I wish meson copied .pyx source to the build dir automatically
```

Comment:
Is this copy actually required? Seems to somewhat blur the lines of the purpose of an in source vs out of source build if we have to copy the source over

Reply:
AFAICT the ones that are passed to the extension_module() definitions below are the original sources, so it's not clear to me either, what this is doing.

In general I would not expect this to be needed unless somehow cython is attempting to automatically figure out information traversing multiple files. For example includes. But there do not seem to be any generated sources in this directory anyway?

Reply:
The .pxi files are generated in the build directory, but the other sources are in the source directory. I don't think cython has an option to pass an include directory for .pxi files, so we have to copy the source files that depend on the .pxi files over. 

If a source doesn't have a dependency on a .pxi file, we don't copy it.

Reply:
What confused me mostly is that you don't seem to be using the copied .pxd anyway? Meson is getting told to run the `cython` tool on the file in the source directory, not the build directory, as far as I can tell.

Reply:
> I don't think cython has an option to pass an include directory for .pxi files

Right, and this is another thing that may make sense to add to cython upstream, there's a couple other options that already got added while SciPy was porting over.

Reply:
> What confused me mostly is that you don't seem to be using the copied .pxd anyway? Meson is getting told to run the `cython` tool on a .pyx in the source directory, not the build directory, as far as I can tell.

cython_sources is a dict mapping source file -> configure_file output(build directory location). 

The copied pxd would be used if we did something like ``cython_sources['algos.pyx']``.

Reply:
Not sure if this is too crazy but would it make sense to just have any imports of the generated templates assume those templates come from the build folder rather than the source?

Reply:
This would mean patching Cython unfortunately. I don't think the include dir command line option works, for pxi files.

Reply:
Yeah I was afraid of that... I wonder why not though. Is this a cython bug? If not, it's definitely a cython feature request.